UKHomeOffice/nmsw-ui
# NMSW-1057 Filter passenger status request

# Ticket

NMSW-1057

----

## AC

As an agent

I want filter on the status responses based on the following

AC - view filter

GIVEN I have received the status responses for a FAL6 
WHEN I view the responses
THEN there is a checkbox filter for board, check and no board

AC - View board only passengers

GIVEN I have received the status responses for a FAL6 
WHEN I filter for passengers with a Board status
THEN  I can see passengers with status 0A (and any other selected checkboxes)

AC - View check only

GIVEN I have received the status responses for a FAL6 
WHEN I filter for passengers with a Board status
THEN  I can  see passengers with status 0B, 0Z, 2Z and 0P (and any other selected checkboxes)

AC - View no board only

GIVEN I have received the status responses for a FAL6 
WHEN I filter for passengers with a "no Board" status
THEN  I see passengers with status 1Z (and any other selected checkboxes)

----

## To test

- run app
- create a declaration
- upload a file with multiple passengers
- (Note: I think all passengers come back with Board status)
- Click on Do not board checkbox
- > You should see no results but you should see correct query in the network tab
- deselect No board and select check documents
- > You should see no results but you should see correct query in the network tab
- deselect No board and select Valid permission
- > You should all results and you should see correct query in the network tab
- keep valid selected and also select check or no board
- > You still see all results and you should see correct query in the network tab
- Search for a name you know is in the file
- > You should see results for the name you selected as well as the filters
- Remove the Valid selection
- > You should no longer see results for the name you searched for
----

## Notes
